### PR TITLE
Refactored tests to separate Unit tests and Integration tests

### LINKED ISSUE #33: Refactor tests to separate Unit tests and Integration Tests and add more Unit tests
As of now all the tests are added to a single file. To scale out better we need to separate unit tests and Integration tests. We also need to add more unit tests

### PR DESCRIPTION
**Pull request by Rohan-cod**
Fixes #33.

### CODE REVIEW

**Review by srgothi92 — COMMENTED** [review]
Thanks for this, I think we would have to wait for other commit to merge before this becaus of formatting conflicts.

**Review by srgothi92 — APPROVED** [review]

**Comment by Rohan-cod** [comment]
Apologies for the commit after the approval. Actually, there were some E501(line too long) errors in test_integration.py which would cause the PEP 8 formatting validation to fail even if it was merged after #39.

**Pull request merged**
